tests/providers/google/cloud/transfers/test_azure_blob_to_gcs.py

```diff
@@ -91,3 +91,37 @@ def test_execute(self, mock_temp, mock_hook_gcs, mock_hook_wasb):
             gzip=GZIP,
             filename=mock_temp.NamedTemporaryFile.return_value.__enter__.return_value.name,
         )
+
+    @mock.patch("airflow.providers.google.cloud.transfers.azure_blob_to_gcs.GCSHook")
+    @mock.patch("airflow.providers.google.cloud.transfers.azure_blob_to_gcs.WasbHook")
+    def test_execute_single_file_transfer_openlineage(self, mock_hook_wasb, mock_hook_gcs):
+        from openlineage.client.run import Dataset
+
+        MOCK_AZURE_ACCOUNT_NAME = "mock_account_name"
+        mock_hook_wasb.return_value.get_conn.return_value.account_name = MOCK_AZURE_ACCOUNT_NAME
+
+        mock_hook_gcs.return_value.upload.side_effect = None  # Prevent actual upload
+        operator = AzureBlobStorageToGCSOperator(
+            wasb_conn_id=WASB_CONN_ID,
+            gcp_conn_id=GCP_CONN_ID,
+            blob_name=BLOB_NAME,
+            container_name=CONTAINER_NAME,
+            bucket_name=BUCKET_NAME,
+            object_name=OBJECT_NAME,
+            filename=FILENAME,
+            gzip=GZIP,
+            impersonation_chain=IMPERSONATION_CHAIN,
+            task_id=TASK_ID,
+        )
+
+        operator.execute(context=None)
```

Comment:
FYI, there is no need to mock hooks and run execute, when we implement the _on_start method, you can simply create the operator and then run the OL method ^^

Reply:
I will test and remove the hooks to keep the test simple. do you think we should add any other test?

Reply:
Not really. Usually we try to test some real case scenarios and edge cases, what different / weird object paths can be passed. I think for Operator that does 1:1 file transfer there is not much that can brake, so the tests must ensure that we get expected OL datasets.

Reply:
Ok, but maybe let's remove the execute and GCS hook mock? If we do not execute, we don't need to mock the upload. Pinging @mobuchowski as he is the one with the power to merge the PR.

Reply:
Yeah, if you don't save hooks - as in, use `self.gcs_hook` in `self.azure_hook` but ephemeral objects in execute I don't see reason to run `execute`

Reply:
@mobuchowski I have removed the the gcs hook and execute() from the test.
Please take a look and approve :)